kernel/kernel-api/src/test/scala/io/delta/kernel/internal/SnapshotManagerSuite.scala

```diff
@@ -220,7 +220,9 @@ class SnapshotManagerSuite extends AnyFunSuite with MockFileSystemClientUtils {
           new MockSidecarParquetHandler(expectedSidecars),
           new MockSidecarJsonHandler(expectedSidecars)),
         Optional.empty(),
-        versionToLoad
+        versionToLoad,
+        Optional.empty(),
```

Comment:
This class tests the specific cases by mocking the Json and Parquet handler.

We need to add tests with a mocked Commitcoordinator with different scenarios:
1) a version is missing from the getCommits
2) commit coordinator throws error
3) commit coordinator returns a backfilled commit

etc..